webservices/common/models/reports.py

```diff
@@ -60,6 +60,32 @@ def treasurer_name(self):
         )
         return name
 
+class AmendmentChainMixin(object):
+
+    @property
+    def most_recent(self):
+        return self.file_number == self.amendment.most_recent_filing
+
+    @hybrid_property
+    def most_recent_filing(self):
+        return self.amendment.most_recent_filing
+
+    @hybrid_property
+    def amendment_chain(self):
+        return self.amendment.amendment_chain
+
+    @property
+    def is_amended(self):
+        return not self.most_recent
+
+    @property
+    def amended_by(self):
+        if len(self.amendment.longest_chain) > 0 and self.is_amended:
+            index = self.amendment.longest_chain.index(self.file_number)
```

Comment:
Ah, it looks like you were commenting on that at the same time as me.  I can move some of the logic into the amendment class, I still think a mixin is valid as well, as it makes it easy to inject the behavior into the receiving class, and allows the class to either use it or override.